### PR TITLE
translate task-structure into en-us

### PR DESCRIPTION
**Pull request by Tianqi-Dotes**
translate task-structure into en-us

### CODE REVIEW

**Comment by CalvinKirs** [comment]
@ptyp @Segun-Ogundipe PTAL.Thx

File: docs/en-us/1.3.4/user_doc/task-structure.md [hunk 1]
```diff
@@ -0,0 +1,1134 @@
...
+    ]
+}
+```
+
+
+
+## MapReduce(MR) Node
+**Node data structure as follow:**
+
+order|parameter name||type|description |notes
```
Comment (Segun-Ogundipe) [inline docs/en-us/1.3.4/user_doc/task-structure.md:341]: `order` should be `No.`

File: docs/en-us/1.3.4/user_doc/task-structure.md [hunk 2]
```diff
@@ -0,0 +1,1134 @@
...
+
+
+## PROCEDURE [stored procedures] Node
+**Node data structure as follow:**
+**Node data example:**
+
+## SPARK Node
+**Node data structure as follow:**
+
+order|parameter name||type|description |notes
```
Comment (Segun-Ogundipe) [inline docs/en-us/1.3.4/user_doc/task-structure.md:241]: `order` should be `No.`

File: docs/en-us/1.3.4/user_doc/task-structure.md [hunk 3]
```diff
@@ -0,0 +1,1134 @@
...
+    "preTasks":[
+
+    ]
+}
+```
+
+
+## Python Node
+**Node data structure as follow:**
+order|parameter name||type|description |notes
```
Comment (Segun-Ogundipe) [inline docs/en-us/1.3.4/user_doc/task-structure.md:426]: `order` should be `No.`

File: docs/en-us/1.3.4/user_doc/task-structure.md [hunk 4]
```diff
@@ -0,0 +1,1134 @@
...
+}
+```
+
+
+
+
+## Flink Node
+**Node data structure as follow:**
+
+order|parameter name||type|description |notes
```
Comment (Segun-Ogundipe) [inline docs/en-us/1.3.4/user_doc/task-structure.md:503]: `order` should be `No.`

File: docs/en-us/1.3.4/user_doc/task-structure.md [hunk 5]
```diff
@@ -0,0 +1,1134 @@
...
+    "preTasks":[
+
+    ]
+}
+```
+
+## HTTP Node
+**Node data structure as follow:**
+
+order|parameter name||type|description |notes
```
Comment (Segun-Ogundipe) [inline docs/en-us/1.3.4/user_doc/task-structure.md:598]: `order` should be `No.`

File: docs/en-us/1.3.4/user_doc/task-structure.md [hunk 6]
```diff
@@ -0,0 +1,1134 @@
...
+    ]
+}
+```
+
+
+
+## DataX Node
+
+**Node data structure as follow:**
+order|parameter name||type|description |notes
```
Comment (Segun-Ogundipe) [inline docs/en-us/1.3.4/user_doc/task-structure.md:685]: `order` should be `No.`

File: docs/en-us/1.3.4/user_doc/task-structure.md [hunk 7]
```diff
@@ -0,0 +1,1134 @@
...
+    "preTasks":[
+
+    ]
+}
+```
+
+## Sqoop Node
+
+**Node data structure as follow:**
+order|parameter name||type|description |notes
```
Comment (Segun-Ogundipe) [inline docs/en-us/1.3.4/user_doc/task-structure.md:775]: `order` should be `No.`

File: docs/en-us/1.3.4/user_doc/task-structure.md [hunk 8]
```diff
@@ -0,0 +1,1134 @@
...
+            "preTasks":[
+
+            ]
+        }
+```
+
+## Condition Branch Node
+
+**Node data structure as follow:**
+order|parameter name||type|description |notes
```
Comment (Segun-Ogundipe) [inline docs/en-us/1.3.4/user_doc/task-structure.md:853]: `order` should be `No.`

File: docs/en-us/1.3.4/user_doc/task-structure.md [hunk 9]
```diff
@@ -0,0 +1,1134 @@
...
+        "test01",
+        "test02"
+    ]
+}
+```
+
+
+## Subprocess Node
+**Node data structure as follow:**
+order|parameter name||type|description |notes
```
Comment (Segun-Ogundipe) [inline docs/en-us/1.3.4/user_doc/task-structure.md:918]: `order` should be `No.`

File: docs/en-us/1.3.4/user_doc/task-structure.md [hunk 10]
```diff
@@ -0,0 +1,1134 @@
...
+
+            ]
+        }
+```
+
+
+
+## DEPENDENT Node
+**Node data structure as follow:**
+order|parameter name||type|description |notes
```
Comment (Segun-Ogundipe) [inline docs/en-us/1.3.4/user_doc/task-structure.md:979]: `order` should be `No.`

**Review by Segun-Ogundipe — CHANGES_REQUESTED** [review]
Hi @Tianqi-Dotes, great job on this PR. I left some comments here.

Regards.

File: docs/en-us/1.3.4/user_doc/task-structure.md [hunk 11]
```diff
@@ -0,0 +1,1134 @@
+
+# Overall Tasks Storage Structure
+Dolphinscheduler persists all created tasks in the table 't_ds_process_definition'.
```
[the remark this answers is not in the document]
Comment (Tianqi-Dotes) [reply docs/en-us/1.3.4/user_doc/task-structure.md:3]: done

File: docs/en-us/1.3.4/user_doc/task-structure.md [hunk 12]
```diff
@@ -0,0 +1,1134 @@
...
+
+The following shows the 't_ds_process_definition' table structure:
+
+
+order | field  | type  |  description
```
[the remark this answers is not in the document]
Comment (Tianqi-Dotes) [reply docs/en-us/1.3.4/user_doc/task-structure.md:8]: done all replacemnets

File: docs/en-us/1.3.4/user_doc/task-structure.md [hunk 13]
```diff
@@ -0,0 +1,1134 @@
...
+-------- | ---------| -------- | ---------
+1|id|int(11)|primary key
+2|name|varchar(255)|define workflow name
```
[the remark this answers is not in the document]
Comment (Tianqi-Dotes) [reply docs/en-us/1.3.4/user_doc/task-structure.md:11]: done

File: docs/en-us/1.3.4/user_doc/task-structure.md [hunk 14]
```diff
@@ -0,0 +1,1134 @@
...
+3|version|int(11)|define workflow version
```
[the remark this answers is not in the document]
Comment (Tianqi-Dotes) [reply docs/en-us/1.3.4/user_doc/task-structure.md:12]: done

File: docs/en-us/1.3.4/user_doc/task-structure.md [hunk 15]
```diff
@@ -0,0 +1,1134 @@
...
+4|release_state|tinyint(4)|define workflow publish status：0 not yet launch, 1 launched
```
[the remark this answers is not in the document]
Comment (Tianqi-Dotes) [reply docs/en-us/1.3.4/user_doc/task-structure.md:13]: done

File: docs/en-us/1.3.4/user_doc/task-structure.md [hunk 16]
```diff
@@ -0,0 +1,1134 @@
...
+5|project_id|int(11)|project id
+6|user_id|int(11)|define workflow owner id
```
[the remark this answers is not in the document]
Comment (Tianqi-Dotes) [reply docs/en-us/1.3.4/user_doc/task-structure.md:15]: done

File: docs/en-us/1.3.4/user_doc/task-structure.md [hunk 17]
```diff
@@ -0,0 +1,1134 @@
...
+7|process_definition_json|longtext|define workflow JSON
```
[the remark this answers is not in the document]
Comment (Tianqi-Dotes) [reply docs/en-us/1.3.4/user_doc/task-structure.md:16]: done

File: docs/en-us/1.3.4/user_doc/task-structure.md [hunk 18]
```diff
@@ -0,0 +1,1134 @@
...
+8|description|text|define workflow description
```
[the remark this answers is not in the document]
Comment (Tianqi-Dotes) [reply docs/en-us/1.3.4/user_doc/task-structure.md:17]: done

File: docs/en-us/1.3.4/user_doc/task-structure.md [hunk 19]
```diff
@@ -0,0 +1,1134 @@
...
+9|global_params|text|global parameters
+10|flag|tinyint(4)|specify workflow availability：0 not available, 1 available
```
[the remark this answers is not in the document]
Comment (Tianqi-Dotes) [reply docs/en-us/1.3.4/user_doc/task-structure.md:19]: done

File: docs/en-us/1.3.4/user_doc/task-structure.md [hunk 20]
```diff
@@ -0,0 +1,1134 @@
...
+11|locations|text|node location info
```
[the remark this answers is not in the document]
Comment (Tianqi-Dotes) [reply docs/en-us/1.3.4/user_doc/task-structure.md:20]: done

File: docs/en-us/1.3.4/user_doc/task-structure.md [hunk 21]
```diff
@@ -0,0 +1,1134 @@
...
+12|connects|text|node connectivity info
+13|receivers|text|receivers
+14|receivers_cc|text|CC receivers
+15|create_time|datetime|create time
+16|timeout|int(11) |timeout
+17|tenant_id|int(11) |tenant id
+18|update_time|datetime|update time
+19|modify_by|varchar(36)|specify users make modification
```
[the remark this answers is not in the document]
Comment (Tianqi-Dotes) [reply docs/en-us/1.3.4/user_doc/task-structure.md:28]: done

File: docs/en-us/1.3.4/user_doc/task-structure.md [hunk 22]
```diff
@@ -0,0 +1,1134 @@
...
+20|resource_ids|varchar(255)|resource ids
+
+In which 'process_definition_json' is the core field, defines task info in DAG and is stored in JSON format.
```
[the remark this answers is not in the document]
Comment (Tianqi-Dotes) [reply docs/en-us/1.3.4/user_doc/task-structure.md:31]: done

File: docs/en-us/1.3.4/user_doc/task-structure.md [hunk 23]
```diff
@@ -0,0 +1,1134 @@
...
+
+Following table describes the common data structure
```
[the remark this answers is not in the document]
Comment (Tianqi-Dotes) [reply docs/en-us/1.3.4/user_doc/task-structure.md:33]: done

File: docs/en-us/1.3.4/user_doc/task-structure.md [hunk 24]
```diff
@@ -0,0 +1,1134 @@
...
+order | field  | type  |  description
+-------- | ---------| -------- | ---------
+1|globalParams|Array|global parameters
+2|tasks|Array|task collections in the workflow[structures of each task type please check following sections]
```
[the remark this answers is not in the document]
Comment (Tianqi-Dotes) [reply docs/en-us/1.3.4/user_doc/task-structure.md:37]: done

File: docs/en-us/1.3.4/user_doc/task-structure.md [hunk 25]
```diff
@@ -0,0 +1,1134 @@
...
+            "value":"${system.biz.date}"
+        }
+    ],
+    "tasks":Array[1],
+    "tenantId":0,
+    "timeout":0
+}
+```
+
+# Detailed Explanation About the Storage Structure of Each Task Type
```
[the remark this answers is not in the document]
Comment (Tianqi-Dotes) [reply docs/en-us/1.3.4/user_doc/task-structure.md:58]: done

File: docs/en-us/1.3.4/user_doc/task-structure.md [hunk 26]
```diff
@@ -0,0 +1,1134 @@
...
+    "preTasks":[
+
+    ]
+}
+
+```
+
+
+## SQL Node
+Execute query/update SQL to specified datasource.
```
[the remark this answers is not in the document]
Comment (Tianqi-Dotes) [reply docs/en-us/1.3.4/user_doc/task-structure.md:136]: done

File: docs/en-us/1.3.4/user_doc/task-structure.md [hunk 27]
```diff
@@ -0,0 +1,1134 @@
...
+
+**Node data structure as follow:**
```
[the remark this answers is not in the document]
Comment (Tianqi-Dotes) [reply docs/en-us/1.3.4/user_doc/task-structure.md:138]: done all replacement

File: docs/en-us/1.3.4/user_doc/task-structure.md [hunk 28]
```diff
@@ -0,0 +1,1134 @@
...
+order|parameter name||type|description |note
+-------- | ---------| ---------| -------- | --------- | ---------
+1|id | |String|task id|
+2|type ||String |task type |SQL
+3| name| |String|task name|
+4| params| |Object|customized parameters|Json format
+5| |type |String |database type
+6| |datasource |Int |datasource id
+7| |sql |String |query SQL sentence
```
[the remark this answers is not in the document]
Comment (Tianqi-Dotes) [reply docs/en-us/1.3.4/user_doc/task-structure.md:147]: done

File: docs/en-us/1.3.4/user_doc/task-structure.md [hunk 29]
```diff
@@ -0,0 +1,1134 @@
...
+8| |udfs | String| udf functions|specify UDF function id, separate by comma
```
[the remark this answers is not in the document]
Comment (Tianqi-Dotes) [reply docs/en-us/1.3.4/user_doc/task-structure.md:148]: done

File: docs/en-us/1.3.4/user_doc/task-structure.md [hunk 30]
```diff
@@ -0,0 +1,1134 @@
...
+9| |sqlType | String| SQL node type |0 for query and 1 for none query SQL
```
[the remark this answers is not in the document]
Comment (Tianqi-Dotes) [reply docs/en-us/1.3.4/user_doc/task-structure.md:149]: done

File: docs/en-us/1.3.4/user_doc/task-structure.md [hunk 31]
```diff
@@ -0,0 +1,1134 @@
...
+    "tasks":Array[1],
+    "tenantId":0,
+    "timeout":0
+}
+```
+
+# Detailed Explanation About the Storage Structure of Each Task Type
+
+## Shell Nodes
+**Node data structure as follow:**
```
[the remark this answers is not in the document]
Comment (Tianqi-Dotes) [reply docs/en-us/1.3.4/user_doc/task-structure.md:61]: done

**Comment by Tianqi-Dotes** [comment]
@Segun-Ogundipe  Hi, Segun
Deeply thanx for reviewing. Have done all the changes.
Best wishes
Tianqi

**Review by CalvinKirs — APPROVED** [review]
LGTM

**Pull request merged**